drone-localization/core/services/evaluation/metrics.py

```
@@ -0,0 +1,83 @@
+    return {
+        "R@1": recall_1 / num_queries,
+        "R@5": recall_5 / num_queries,
+        "R@10": recall_10 / num_queries,
+        "mAP": np.mean(all_ap)
+    }
+
+def evaluate_model(model, mode, root_data):
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    model.eval()
+
+    query_dataset = SUESDataset(root_dir=root_data, transform=get_transforms(mode, is_train=False), split='test', view='drone')
```

Ruff (0.15.12)

[error] Undefined name `SUESDataset`

(F821)